app/models/ext_management_system.rb

```diff
@@ -300,18 +300,19 @@ def connection_configurations
   # hostname, port, and authentication
   # if no role is passed in assume is default role
   def add_connection_configuration_by_role(options)
-    unless options[:endpoint].key?(:role)
-      options[:endpoint][:role] ||= "default"
+    connection = options.deep_symbolize_keys
```

Comment:
I like the renaming of the param to a name with documentation value.
Let's also do this with the named param in the method signature
```ruby
def add_connection_configuration_by_role(connection)
  connection = connection.deep_symbolize_keys
```